src/main/webapp/app/overview/discussion-section/discussion-section.component.ts

```diff
@@ -90,28 +92,64 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
         this.postCreateEditModal?.modalRef?.close();
     }
 
+    /**
+     * on changing the sort direction via icon, the metis service is invoked to deliver the posts for the currently set context,
+     * sorted on the backend
+     */
+    onChangeSortDir(): void {
+        switch (this.currentSortDirection) {
+            case undefined: {
+                this.currentSortDirection = SortDirection.ASCENDING;
+                break;
+            }
+            case SortDirection.ASCENDING: {
+                this.currentSortDirection = SortDirection.DESCENDING;
+                break;
+            }
+            default: {
+                this.currentSortDirection = undefined;
+                break;
+            }
+        }
+        this.posts.sort(this.sectionSortFn);
+    }
+
     /**
      * sorts posts by following criteria
      * 1. criterion: displayPriority is PINNED -> pinned posts come first
      * 2. criterion: displayPriority is ARCHIVED  -> archived posts come last
      * -- in between pinned and archived posts --
-     * 3. criterion: vote-emoji count -> posts with more vote-emoji counts comes first
-     * 4. criterion: creationDate -> most recent comes at the end (chronologically from top to bottom)
+     * 3. criterion: creationDate - if activated by user through the sort arrow -> most recent comes at the end (chronologically from top to bottom)
+     * 4. criterion: if 3'rd criterion was not activated by the user, vote-emoji count -> posts with more vote-emoji counts comes first
+     * 5. criterion: if 3'rd criterion was not activated by the user, most recent posts comes at the end (chronologically from top to bottom)
      * @return Post[] sorted array of posts
      */
-    sectionSortFn(postA: Post, postB: Post): number {
+    sectionSortFn = (postA: Post, postB: Post): number => {
+        // 1'st criterion
         if (postA.displayPriority === DisplayPriority.PINNED && postB.displayPriority !== DisplayPriority.PINNED) {
             return -1;
         }
         if (postA.displayPriority !== DisplayPriority.PINNED && postB.displayPriority === DisplayPriority.PINNED) {
             return 1;
         }
+
+        // 2'nd criterion
         if (postA.displayPriority === DisplayPriority.ARCHIVED && postB.displayPriority !== DisplayPriority.ARCHIVED) {
             return 1;
         }
         if (postA.displayPriority !== DisplayPriority.ARCHIVED && postB.displayPriority === DisplayPriority.ARCHIVED) {
             return -1;
         }
+
+        // 3'rd criterion
+        if (!!this.currentSortDirection) {
+            const comparison = this.sortByDate(postA, postB);
+            if (comparison !== 0) {
+                return comparison;
+            }
+        }
+
+        // 4'th criterion
```

Comment:
```suggestion
        // 4th criterion
```
Nitpick (same as above)